temperature/router.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import APIRouter, Depends, Query, HTTPException
+from sqlalchemy.orm import Session
+
+from dependencies import get_db
+from temperature import crud, schemas
+
+
+router = APIRouter()
+
+
+@router.get(
+    "/temperature/",
+    response_model=list[schemas.Temperature]
+)
+async def get_temperature_list(
+        city_id: int = Query(None, description="City id"),
+        db: Session = Depends(get_db)
+):
+
+    if city_id:
+        temperature = await crud.get_temperatures_by_city_id(
+            db=db, city_id=city_id
+        )
+        if not temperature:
+            raise HTTPException(status_code=400, detail="Temperature does not exist")
+
+        return [temperature]
+
+    return await crud.get_temperature_list(db=db)
+
+
+@router.post("/temperature/update/", response_model=list[schemas.Temperature])
```

Comment:
The `update_temperature` endpoint is defined with a response model of `list[schemas.Temperature]`, but the `set_temperature` function in `crud.py` returns a `JSONResponse` with a message. Consider aligning the response model with the actual response or adjusting the function to return a list of `Temperature` objects.